deepparse/app/api.py

```diff
@@ -0,0 +1,155 @@
+"""REST API."""
+from typing import List, Dict, Any
+from contextlib import asynccontextmanager
+
+from enum import Enum
+
+
+try:
+    from pydantic import BaseModel, Field
+    from fastapi import FastAPI, Depends
+    from fastapi.responses import JSONResponse
+    from fastapi.staticfiles import StaticFiles
+    import uvicorn
+    from deepparse.download_tools import MODEL_MAPPING_CHOICES, download_models
+    from deepparse.parser import AddressParser
+    from deepparse.app.deepparser_logger import logger
+    from deepparse.app.sentry import configure_sentry
+    import os
+
+
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError("Ensure you installed the extra packages using: 'pip install deepparse[app]'") from e
+
+
+address_parser_mapping: Dict[str, AddressParser] = {}
+
+
+class ModelName(str, Enum):
+    bpemb = "bpemb"
+    bpemb_attention = "bpemb-attention"
+    fasttext = "fasttext"
+    fasttext_attention = "fasttext-attention"
+    fasttext_light = "fasttext-light"
+
+
+class AddressInput(BaseModel):
+    address: str = Field(examples=["350 rue des Lilas Ouest Quebec city Quebec G1L 1B6"])
+
+
+def initialize_address_parser_mapping() -> Dict[str, AddressParser]:
+    _address_parser_mapping: Dict[str, AddressParser] = {}
+    logger.debug("Downloading models")
+    download_models()
+    for model in MODEL_MAPPING_CHOICES:
+        if model == "bmepb":
+            logger.debug("initializing %s", model)
+            attention = False
+            if "-attention" in model:
+                attention = True
+            _address_parser_mapping[model] = AddressParser(
+                model_type=model,
+                offline=True,
+                attention_mechanism=attention,
+                device="cpu",
+            )
```

Comment:
This if statement makes it so that only the `bpemb` value works. So `bpemb-attention`, `fasttext`, etc, raise a server error because a KeyError is raised in [line 65](https://github.com/GRAAL-Research/deepparse/blob/7337538577067e89cb20bcbaa00d088f5022d6eb/deepparse/app/api.py#L65).